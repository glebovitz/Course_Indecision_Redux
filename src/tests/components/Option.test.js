import React from 'react';
import { shallow } from 'enzyme';
import Option from '../../components/Option';

test('should render an Option Component correctly', () => {
  const wrapper = shallow(<Option />);
  expect(wrapper).toMatchSnapshot();
});

test('should render an Option Component with properties correctly', () => {
  const wrapper = shallow(<Option count={5} optionText={'Test Text'}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should call deleteOption on remove button click', () => {
  const value = 'abcdef12345';
  const deleteOption = jest.fn();
  const wrapper = shallow(<Option deleteOption={deleteOption} optionText={value}/>);
  wrapper.find('button').simulate("click");
  expect(deleteOption).toHaveBeenLastCalledWith(value);
});
