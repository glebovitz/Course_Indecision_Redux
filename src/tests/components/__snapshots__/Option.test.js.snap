// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an Option Component correctly 1`] = `
<div
  className="option"
>
  <p
    className="option__text"
  >
    . 
  </p>
  <button
    className="button button--link"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`should render an Option Component with properties correctly 1`] = `
<div
  className="option"
>
  <p
    className="option__text"
  >
    5
    . 
    Test Text
  </p>
  <button
    className="button button--link"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
